### ISSUE TITLE
Fronted discussion

### ISSUE THREAD

**Issue by amis92** [BSData/bsdata#228]
The website is currently written in Angular.js. It's an outdated framework, and there will only be less and less people able to help us fix it.

As such it's probably best to rewrite the app in a more modern framework with considerable support and community.

**Comment by NistrumCain** [comment]
If you want me to take a look at the frontend code it would be really good if i could get a good idea of what features/changes/bugs you want/need/have for the platform going forward.

**Comment by amis92** [comment]
To be honest, the frontend is not heavily used anymore, except the "Report a bug" feature.

So, the features currently required for frontend:
* display a list of repositories
* display a list of datafiles in repository
* report-a-bug feature as a popup or something else

There are a couple of static links as well, and there's a feature to anonymously provide a datafile update. But it's being rarely used, and I don't think much will be lost by just dropping it, it's also the most complicated part of the current feature set.

Going forward, it'd probably be nice to include a search/filter, since we're now in vicinity of 100 repos. There are some talks about localizations of datafiles, and maybe channels for repos, like alpha/beta/release etc.

**Comment by NistrumCain** [comment]
Yeah all of that is simple enough, itll take me longer to solve this java problem than it will to write that lot up

**Comment by amis92** [comment]
Well, if you need to choose, I think the fronted will be much more valuable contribution :)

**Comment by NistrumCain** [comment]
Sure I'll build a new frontend and I can look at getting that pesky backend to work once I'm a little more familiar with the page..

**Comment by NistrumCain** [comment]
Sure I'll build a new frontend and I can look at getting that pesky backend to work once I'm a little more familiar with the page..

**Comment by NistrumCain** [comment]
Do you want to keep the look the same or not?

**Comment by amis92** [comment]
I don't think we need to. If you have any better/more modern ideas, I'm very open :)

**Comment by amis92** [comment]
Also, the logo could be updated to the one Grotskull's Workshop from Discord prepared from us:

Pics from https://discordapp.com/channels/558412685981777922/558412685981777924/560073806291664936

![logo round](https://cdn.discordapp.com/attachments/558412685981777924/560073910813589524/Icon_Round.jpg)

![logo square](https://cdn.discordapp.com/attachments/558412685981777924/560073805645479937/Icon_Rework1.jpg)

 ![logo less busy](https://cdn.discordapp.com/attachments/558412685981777924/560074118364528650/Icon_Rework3.jpg)

**Comment by amis92** [comment]
Closed by accident

**Comment by NistrumCain** [comment]
I was wondering about that! Well as for fe design stuff I can try and make something that fits the newer logo a little more maybe... we will see

**Comment by NistrumCain** [comment]
hey, didnt mean to vanish for a month, set up a local project and started to sort the code out, what do you want me to do about a repo, i can set up a personal one until im ready to launch/ handover but what should i do when its all done?

**Comment by amis92** [comment]
Do a fork, work on the fork, and open a PR. The sooner you open, the sooner we can discuss the rewrite.
 At least that's what I'd suggest.

**Comment by amis92** [comment]
Also, glad you didn't vanish forever ;D

**Comment by NistrumCain** [comment]
Yeah, will do in the next few days hopefully. My plan right now is to build the site to be identical and then make incremental changes

**Comment by amis92** [comment]
Cool. Though, don't spend too much time on making it "identical" initially ;)

**Comment by NistrumCain** [comment]
Haha, I take it there is still no one who can help with the java call? I might be able to get someone to look at it?

**Comment by amis92** [comment]
Wait, I've forgotten - what java problem are you referring to?

Sure, if you've got someone that can fix this - by all means, the more the merrier. :)

**Comment by NistrumCain** [comment]
The problem with the api not always returning all of the repos

**Comment by amis92** [comment]
Right. Yeah. This is not the priority here, #208 will be fixed with a rewrite of the backend using webhooks instead of long polling.

**Comment by NistrumCain** [comment]
Awesome in that case I wont worry about it.

**Comment by amis92** [comment]
@NistrumCain I know it's probably not very nice of me, but: any progress? Are you still in? :)

**Comment by NistrumCain** [comment]
Sorry man, still here! I even check this thread once in a while. I am off work for a bit soon in the next 2 weeks and was planning to work on it all then. Dont feel bad for bugging me, i have had plenty of time.. its my fault not yours!

**Comment by amis92** [comment]
Soo let's push a PR already so that there's *something* visible happening :>

**Comment by NistrumCain** [comment]
hahah, i am literally sat working on it now. i might push a pr when im done but itll be really WIP

**Comment by NistrumCain** [comment]
there was a lot of chasing information around before i got to the bottom of where everything connects together. my plan right now is to rip out all of the webapp code and replace it with a react app.  I will make a branch and start a basic app and at least if i have pushed something into a branch we will have some progress

**Comment by amis92** [comment]
Cool. Don't overthink stuff. It's a pretty static page. Great to know you're still on it! :D

**Comment by NistrumCain** [comment]
I've got a non breaking pr set up.. I've hidden  my setup is inside the web app folder you can easily get it to run

**Comment by bguyan** [comment]
So, I have a question. If you switch this whole system over, will it still be using the same hosting system?

**Comment by amis92** [comment]
@bguyan you mean will we still use AppSpot? Maybe, maybe not. Hasn't been decided yet. Why?

**Comment by bguyan** [comment]
i may or may not be nistrum...

was wondering if we were thinking of using something like AWS? or google cloud?

**Comment by amis92** [comment]
AppSpot *is* GoogleCloud. 🤔

**Comment by bguyan** [comment]
3 guesses what system i am used to! I knew that appSpot was on googles system, im just not used to their serverless hosting platforms..

**Comment by amis92** [comment]
AWS?

Cool. Yeah, since we're attempting a rewrite of backend in .NET, there was some discussion about moving to Azure, but it's all just talking.

**Comment by bguyan** [comment]
the reason i asked is because i feel like if you are doing a new BE and i have sorted a new FE we could publish the whole lot together and sidestep the current system? it kind of changes our publication pipeline

**Comment by bguyan** [comment]
yeah, I work a lot on AWS. i do all my publication through the AWS platform

**Comment by amis92** [comment]
To be honest, with all the dynamic we have here, I don't see us moving anywhere. So *if* we actually get both -ends done at some point, we'll go straight to where we have *something* which is AppSpot.

If you have a better suggestion, shoot away. We're currently using about 10 TB bandwidth per month.

**Comment by bguyan** [comment]
yeah, thats fair, good to know, makes life simpler!

**Comment by amis92** [comment]
I mean, it's possible to move - I just don't have enough data to decide which hosting would be cheaper/st, and also it's not me who pays the bill sooo. :D

**Comment by bguyan** [comment]
nah, things are set up, im just not sure how easy it is to publish on the platform,  i just need to do some reading

**Comment by amis92** [comment]
Well I'd prefer for it to be easier ;D

**Comment by bguyan** [comment]
haha, its a bigger issue for you than me i think? all i need to do is drop my code in to replace the current index.html and my JS will do the rest :P 

i might get some more dev time on this on thursday... will see. I have to work on setting up my own site for my app soon to ill try and connect the 2 so i spend some time on the BS frontend any time i work on my own
